src/main/scala/com/evobootcamp/homeworks/basics/Collections.scala

```diff
@@ -0,0 +1,132 @@
+      acc match {
+        case (l, r) => v match {
+          case "R"  => (l + 1, if (l + 1 == 0) r + 1 else r)
+          case "L"  => (l - 1, if (l - 1 == 0) r + 1 else r)
+        }
+      }
+    })._2
+  }
+
+  // https://leetcode.com/problems/matrix-block-sum/
+  def matrixBlockSum(mat: Array[Array[Int]], K: Int): Array[Array[Int]] = {
+    val matZipped = mat.zipWithIndex
+    matZipped.map {
+      case (mt1, i1) => {
+        val iRange1 = i1 - K to i1 + K
+
+        mt1.zipWithIndex.map({
+          case (_, i2) => {
+            val iRange2 = i2 - K to i2 + K
+
+            matZipped.filter({
+              case (_, idx) =>  iRange1 contains idx
+            }).flatMap({
+              case (m2, _) => {
+                m2.zipWithIndex.filter({
+                  case (_, idx) => iRange2 contains idx
+                }).map {
+                  case (v, _) => v
+                }
+              }
+            }).sum
+          }
+        })
+      }
+    }
+  }
+
+  // Implement scanLeft (not using scans ofc)
+  def scanLeft[T](zero: T)(list: List[T])(f: (T, T) => T): List[T] = {
+    @tailrec
+    def recursive(list: List[T], result: List[T], zero: T): List[T] = list match {
+      case Nil  => result
+      case x :: xs  => {
+        val a = f(zero, x)
+        recursive(xs, result.appended(a), a)
+      }
+    }
+
+    recursive(list, List(zero), zero)
+  }
+
+  // https://twitter.com/allenholub/status/1357115515672555520/photo/1
+  // pass the interview
+  def count(s: String): List[(Char, Int)] = {
+    @tailrec
+    def recursive(result: List[(Char, Int)], list: List[String], char: String, count: Int): List[(Char, Int)] = list match {
+      case Nil  => if (count > 0) result.appended((char.charAt(0), count)) else result
+      case x :: xs  => {
+        recursive(
+          if (char != x && count > 0) result.appended((char.charAt(0), count)) else result,
+          xs,
+          x,
+          if (char != x) 1 else count + 1
+        )
+      }
+    }
+
+    recursive(List(), s.split("").toList, "", 0)
+  }
+
+  // Implement a special sort which sorts the keys of a map (K) according to their associated
+  // values (V).
+  //
+  // In case of "ties" (equal values) it should group these keys K into Set-s in the results.
+  //
+  // The input is a map where keys (K) are the values to be sorted and values are their associated numeric
+  // values.
+  //
+  // The output is a list (in ascending order according to the associated `Int` values) of tuples of `Set`-s
+  // with values from K, and the associated value V for these values in the `Set`.
+  //
+  // For example:
+  //
+  // Input `Map("a" -> 1, "b" -> 2, "c" -> 4, "d" -> 1, "e" -> 0, "f" -> 2, "g" -> 2)` should result in
+  // output `List(Set("e") -> 0, Set("a", "d") -> 1, Set("b", "f", "g") -> 2, Set("c") -> 4)`.
+  def sortConsideringEqualValues[T](map: Map[T, Int]): List[(Set[T], Int)] = map.groupMap({
+    case (_, c) => c
+  })({
```

Comment:
Try not to use both parentheses at the same time. If you need {} then it is better to remove ()